components/progress-bar/src/ProgressBar.tsx

```diff
@@ -0,0 +1,159 @@
+  progressContainerClasses?: string | CSSProperties;
+  progressText?: string;
+  progressTextClasses?: string | CSSProperties;
+  radius?: number;
+  circleRadiousInPercentage?: string;
+  enableCircularProgress?: boolean;
+  StrokeColour?: string;
+  progressColour?: string;
+}
+
+const ProgressBar = ({
+  progressClasses,
+  progressContainerClasses,
+  progressInterval,
+  running,
+  defaultProgress,
+  totalFileSize = 0,
+  processedFileSize = 0,
+  progressText,
+  progressTextClasses,
+  radius = 95,
+  circleRadiousInPercentage,
+  enableCircularProgress,
+  StrokeColour,
+  progressColour,
+}: ProgressBarInterface) => {
+  const defaultProgressClasses = "flex h-full rounded-full bg-green-600";
+  const defaultProgressContainerClasses =
+    "flex h-2 rounded-md mx-1 w-full rounded-full overflow-hidden";
+
+  const [progressStyle, setProgressStyle] = useState({});
+  const [statusStyle, setStatusStyle] = useState({});
+
+  const { progress } = useProgress({
+    progressInterval,
+    running,
+    defaultProgress,
+    totalFileSize,
+    processedFileSize,
+  });
+
+  useEffect(() => {
+    const circumference = 2 * Math.PI * radius;
+    const offset = circumference - (progress / 100) * circumference;
+
+    setProgressStyle({
+      strokeDasharray: `${circumference} ${circumference}`,
+      strokeDashoffset: offset,
+    });
+
+    setStatusStyle({
+      strokeDasharray: `${circumference} ${circumference}`,
+      strokeDashoffset: 0,
+    });
+  }, [progress]);
+
+  return (
+    <>
+      {enableCircularProgress ? (
+        <div
+          className={`h-full w-full flex justify-center items-center ${
+            typeof progressContainerClasses === "string"
+              ? twMerge(
+                  defaultProgressContainerClasses,
+                  progressContainerClasses
+                )
+              : defaultProgressContainerClasses
+          }`}
+          style={
+            typeof progressContainerClasses === "object"
+              ? progressContainerClasses
+              : {}
+          }
+        >
+          <svg
+            className={`w-52 h-52 rounded-full ${
+              typeof progressClasses === "string" ? progressClasses : ""
+            }`}
+          >
+            <circle
+              className={`fill-transparent stroke-[10] -rotate-90 origin-center stroke-gray-300 ${StrokeColour}`}
+              cx="50%"
+              cy="50%"
+              r={circleRadiousInPercentage ? circleRadiousInPercentage : "45%"}
+              style={statusStyle}
+            ></circle>
+            <circle
+              className={`fill-transparent stroke-[10] -rotate-90 origin-center stroke-green-500 ${progressColour}`}
+              cx="50%"
+              cy="50%"
+              r={circleRadiousInPercentage ? circleRadiousInPercentage : "45%"}
+              style={progressStyle}
+            ></circle>
+          </svg>
+          <span className="absolute text-2xl font-bold">
```

Comment:
Should these values be static or should we create a prop to pass them along from user, and use default values for this if no classname/CSS properties are given? Should this always be shown for circular Progress bar?

Reply:
Now it is Handle by the Function, user is able to handle Progress Text and progress Value.